src/imageupload/imageuploadediting.js

```diff
@@ -192,6 +193,34 @@ export default class ImageUploadEditing extends Plugin {
 				const viewImg = viewFigure.getChild( 0 );
 				const promise = loader.upload();
 
+				// Force re–paint in Safari. Without it, the image will display with a wrong size.
+				// https://github.com/ckeditor/ckeditor5/issues/1975
+				/* istanbul ignore next */
+				if ( env.isSafari ) {
+					editor.editing.view.once( 'render', () => {
+						// Early returns just to be safe. There might be some code ran
+						// in between the outer scope and this callback.
+						if ( !viewImg.parent ) {
+							return;
+						}
+
+						const domFigure = editor.editing.view.domConverter.mapViewToDom( viewImg.parent );
+
+						if ( !domFigure ) {
+							return;
+						}
+
+						const originalDisplay = domFigure.style.display;
+
+						domFigure.style.display = 'none';
+
+						// Make sure this line will never be removed during minification for having "no effect".
+						domFigure._ckHack = domFigure.offsetHeight;
```

Comment:
Comment for a future me – this is purely theoretical case today that this may be removed during minification. That'd have to be a really eager minifier which would be removing things that don't have any side effect. Perhaps even impossible to use in practice today due to code which actually use such constructs to leave some effects. But I imagined my future self trying to debug why this hack for Safari doesn't work in a minified code only and with this specific minifier and I thought – noooo, I won't do that to myself :D